### ISSUE TITLE
Can't submit a training class registration

### ISSUE THREAD

**Issue by stanleymurgolo** [NIHL/Drupal7-Migration#146]
Trying to submit a training class registration you get the following Technical Difficulties error

The website that you're trying to reach is having technical difficulties and is currently unavailable.

We are aware of the issue and are working hard to fix it. Thank you for your patience.

Using Firefox 45.0.2

**Comment by tgifford-webfirst** [comment]
This may have been due to code being pushed out to fix another issue. Please try again.

**Comment by stanleymurgolo** [comment]
Just tried didn't work, can try again later.

**Comment by tgifford-webfirst** [comment]
I was able to register as an anonymous user here: http://nihlibrary.prod.acquia-sites.com/node/70056/register

Are you logged into drupal, logged in via ezproxy, or completely anonymous when testing?

**Comment by stanleymurgolo** [comment]
I tried logged into drupal and anonymous at that link and got technical difficulties page.

**Comment by stanleymurgolo** [comment]
Also as a side question if you registered at that link, it still shows 20 seats left?

**Comment by tgifford-webfirst** [comment]
I deleted my registration, it showed 19 while I was registered.

The good news is that I was able to reproduce the issue after trying it again, so I should be able to track down the issue now.

**Comment by tgifford-webfirst** [comment]
There appears to have been some data corruption at some point in the past. There were no records in the registration table, but there were records in the associated field tables (e.g. for first and last name, organization, etc.). I cleared out the field tables, so registration should work now. 

The reason why my earlier registration worked is that there were gaps in the data in the field tables, so I happened to fill in a gap; the error only occurred if you had a key collision with existing data.
